packages/lib/modules/pool/migrations/PoolMigrationModal.tsx

```diff
@@ -0,0 +1,77 @@
+import { Modal, ModalBody, ModalCloseButton, ModalContent, ModalOverlay } from '@chakra-ui/react'
+import { getStylesForModalContentWithStepTracker } from '../../transactions/transaction-steps/step-tracker/step-tracker.utils'
+import { useBreakpoints } from '@repo/lib/shared/hooks/useBreakpoints'
+import { useRemoveLiquidity } from '../actions/remove-liquidity/RemoveLiquidityProvider'
+import { DesktopStepTracker } from '../../transactions/transaction-steps/step-tracker/DesktopStepTracker'
+import { usePoolRedirect } from '../pool.hooks'
+import { useOnUserAccountChanged } from '../../web3/useOnUserAccountChanged'
+import { TransactionModalHeader } from '@repo/lib/shared/components/modals/TransactionModalHeader'
+import { useRemoveLiquidityReceipt } from '../../transactions/transaction-steps/receipts/receipt.hooks'
+import { Pool, ProtocolVersion } from '../pool.types'
+import { useUserAccount } from '../../web3/UserAccountProvider'
+import { ActionModalFooter } from '@repo/lib/shared/components/modals/ActionModalFooter'
+import { MigrateLiquiditySummary } from './MigrateLiquiditySummary'
+import { useMigrateLiquidity } from './MigrateLiquidityProvider'
+import { GqlChain } from '@repo/lib/shared/services/api/generated/graphql'
+
+export function PoolMigrationModal() {
+  const { isDesktop } = useBreakpoints()
+  const { userAddress } = useUserAccount()
+  const { oldPool, migrationSteps } = useMigrateLiquidity()
+  const { hasQuoteContext, removeLiquidityTxHash, lastTransaction, urlTxHash } =
+    useRemoveLiquidity()
+
+  const { redirectToPoolPage } = usePoolRedirect(oldPool as Pool)
+  useOnUserAccountChanged(redirectToPoolPage)
+
+  const removeLiquidityReceipt = useRemoveLiquidityReceipt({
+    chain: oldPool?.chain || GqlChain.Mainnet,
+    txHash: removeLiquidityTxHash,
+    userAddress,
+    protocolVersion: oldPool?.protocolVersion as ProtocolVersion,
+    txReceipt: lastTransaction?.result,
+  })
+
+  const isSuccess = !!removeLiquidityTxHash && removeLiquidityReceipt.hasReceipt // FIXME: [JUANJO] just check last tx?
```

Comment:
ya last tx (add liquidity in this case) is the legacy pattern which seems to work okay

Reply:
Seems like it comes from this line `  const quoteAmountsOut = simulationQuery.data?.amountsOut || emptyTokenAmounts(pool)`  on `RemoveLiquidityProvider`. The empty token amounts is what we have here. Not sure what is their purpose and why are they not just an empty array when no info yet (`pool.poolTokens` returns the BPT as well as the pool tokens).

Reply:
Changed by using the actionable tokens as the empty state.